hypertension/dc-7101-algorithms/py-root/test/main_test.py

```diff
@@ -1,14 +1,214 @@
+                }
+            }
+        ),
+        # sufficient_to_autopopulate returns 'success': False, but history_of_diastolic_bp doesn't
+        # Note that the inverse can't happen (where history_of_diastolic_bp fails while sufficient_to_autopopulate doesn't)
+        # because the only way history_of_diastolic_bp can fail is if there are no bp readings, which would cause
+        # sufficient_to_autopopulate to fail as well 
+        (
+            {
+                "bp": [
+                    {
+                        "diastolic": 115,
+                        "systolic": 180,
+                        "date": "2020-11-01"
+                    },
+                    {
+                        "diastolic": 110,
+                        "systolic": 200,
+                        "date": "2020-09-01"
+                    }
+                ],
+                "medication": [],
+                'date_of_claim': '2021-11-09',
+                'veteran_is_service_connected': True
+            },
+            {
+                "statusCode": 209,
+                "headers": {
+                    "Access-Control-Allow-Headers" : "Content-Type",
+                    "Access-Control-Allow-Origin": "*",
+                    "Access-Control-Allow-Methods": "OPTIONS,POST"
+                },
+                "body": {
+                    "predominance_calculation": {
+                        "success": False,
+                    },
+                    "diastolic_history_calculation": {
+                        "diastolic_bp_predominantly_100_or_more": True,
+                        "success": True 
+                    },
+                    "requires_continuous_medication": {
+                        "continuous_medication_required": False,
+                        "success": True
+                    }
+                }
+            }
+        ),
+        # sufficiency and history algos fail
+        (
+            {
+                "bp": [
+                ],
+                "medication": [],
+                'date_of_claim': '2021-11-09',
+                'veteran_is_service_connected': True
+            },
+            {
+                "statusCode": 400,
+                "headers": {
+                    "Access-Control-Allow-Headers" : "Content-Type",
+                    "Access-Control-Allow-Origin": "*",
+                    "Access-Control-Allow-Methods": "OPTIONS,POST"
+                },
+                "body": {
+                    "predominance_calculation": {
+                        "success": False,
+                    },
+                    "diastolic_history_calculation": {
+                        "success": False 
+                    },
+                    "requires_continuous_medication": {
+                        "continuous_medication_required": False,
+                        "success": True
+                    }
+                }
+            }
+        ),
+        # Bad data (KeyError)
```

Comment:
I'm having a hard time understanding what's wrong here. Can we chat about it?

Reply:
A KeyError is thrown because I'm missing the "diastolic" key in the second reading, and have a key "medication" in it's place (though, now that I'm thinking about it, I should probably remove the "medication" key, given the error is caused by the lack of diastolic key). See the discussion about the 500 error above, it might be that we need to change/remove these tests

Reply:
add more descriptive comment